File: 406-queue-reconstruction-by-height/406-queue-reconstruction-by-height.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        # sort by decreasing height and if height is equal then compare by the number of people increasing
        #It will ensure that for largest people exact k people infront.
        
        people.sort(key=lambda p: (-p[0], p[1]))
        #print(people)
        res = []
        for p in people:
            res.insert(p[1], p)
        return res
        

File: 406-queue-reconstruction-by-height/406-queue-reconstruction-by-height.cpp
class Solution {
public:
    static bool cmp(vector<int> &p1,vector<int> &p2){
        if(p1[0]==p2[0]){
            return p1[1]<p2[1];
        }
        return p1[0]>p2[0];
        
    }
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        vector<vector<int>> res;
        sort(people.begin(),people.end(),cmp);
        for(int i=0;i<people.size();i++){
            res.insert(res.begin()+people[i][1],people[i]);
        }
        return res;
    }
};